plugins/inventory/iocage.py

```diff
@@ -286,6 +308,50 @@ def get_inventory(self, path):
 
                 self.get_properties(t_stdout, results, hostname)
 
+        if hooks:
+            cmd_get_pool = cmd.copy()
+            cmd_get_pool.append(self.IOCAGE)
+            cmd_get_pool.append('get')
+            cmd_get_pool.append('--pool')
+            try:
+                p = Popen(cmd_get_pool, stdout=PIPE, stderr=PIPE, env=my_env)
+                stdout, stderr = p.communicate()
+                if p.returncode != 0:
+                    raise AnsibleError(
+                        f'Failed to run cmd={cmd_get_pool}, rc={p.returncode}, stderr={to_native(stderr)}')
+                try:
+                    iocage_pool = to_text(stdout, errors='surrogate_or_strict').strip()
+                except UnicodeError as e:
+                    raise AnsibleError(f'Invalid (non unicode) input returned: {e}') from e
+            except Exception as e:
+                raise AnsibleError(f'Failed to get pool: {e}') from e
+
+            for hostname, host_vars in results['_meta']['hostvars'].items():
+                iocage_hooks = []
+                for hook in hooks:
+                    path = "/" + iocage_pool + "/iocage/jails/" + hostname + "/root" + hook
+                    cmd_cat_hook = cmd.copy()
+                    cmd_cat_hook.append('cat')
+                    cmd_cat_hook.append(path)
+                    try:
+                        p = Popen(cmd_cat_hook, stdout=PIPE, stderr=PIPE, env=my_env)
+                        stdout, stderr = p.communicate()
+                        if p.returncode != 0:
+                            iocage_hooks.append('-')
+                            continue
+
+                        try:
+                            iocage_hook = to_text(stdout, errors='surrogate_or_strict').strip()
+                        except UnicodeError as e:
+                            raise AnsibleError(f'Invalid (non unicode) input returned: {e}') from e
+
+                    except Exception:
+                        iocage_hooks.append('-')
```

Comment:
Same here.

Also, why not using `None` instead of `-`? `None` can never appear as a real value, so you can distinguish "error happened" from "`-` was actually read".

Reply:
The dash "-" is used in *iocage* to represent a missing value. See for example [ioc_list.py#L259](https://github.com/iocage/iocage/blob/master/iocage_lib/ioc_list.py#L259) or [ioc_list.py#L276](https://github.com/iocage/iocage/blob/master/iocage_lib/ioc_list.py#L276). We've already used it too:

```python
if iocage_ip4_dict['ip4']:
    iocage_ip4 = ','.join([d['ip'] for d in iocage_ip4_dict['ip4']])
else:
    iocage_ip4 = '-'
```